### PR TITLE
Feat: skip full-tree diff when caller knows which paths changed

### PR DESCRIPTION
**Pull request by ledancs**
**Long story short**

When the caller knows which field changed, we skip the full-tree diff and instead expand only the changed path into its ancestor prefixes (filtered by the dependency registry). Subsequent cascading passes still use the full diff for correctness. This should cut down lag drastically as traversal of large models can become expensive quite quickly.

**Skip full-tree diff when caller knows which paths changed**

When updateModel is called after a user edits a single form field, Yavl currently diffs the entire data tree against the previous state to find what changed (`getChangedData`). For large forms, this recursive traversal is expensive and redundant -- the caller already knows which path the user edited.

This change lets callers optionally pass `changedPaths` to `updateModel`. When provided, the first pass of processModelChanges skips the full-tree diff entirely and instead expands the given leaf paths into all ancestor prefixes filtered by `fieldsWithDependencies`.
Subsequent passes (triggered by cascading computed values) still use the full `getChangedData` diff to catch any side effects, so correctness is preserved.

What changed:

* `updateModel` accepts two new optional arguments: `isEqualFn` (4th, same as before) and `changedPaths` (5th). The existing signature is fully backward compatible.
* `processModelChanges` receives the optional `changedPaths` and on pass 0, delegates to the new `expandChangedPaths` helper instead of `getChangedData`.
* `expandChangedPaths` takes the leaf paths (e.g., `['model', 0, 'fields', 'name']`), generates all prefixes (root, model, model[0], ..., the leaf itself), strips array indices for the `fieldsWithDependencies` lookup, deduplicates, and filters by the same `shouldIncludeField` logic that `getChangedData` uses.
* `shouldIncludeField` in `getChangedData.ts` is now exported so `expandChangedPaths` can reuse it.

What didn't change:

* Callers that don't pass `changedPaths` get the exact same behavior as before. No existing API contract is broken.
* Pass > 0 (cascading computed value changes) always uses the full `getChangedData` diff, so computed cascades are handled correctly.
* `validateModel` and all other public APIs are unaffected.

Why this helps:

For a large model, `getChangedData` can consume ~25% of per-keystroke CPU time by walking the entire tree on every edit. With `changedPaths`, that first-pass cost drops to `O(depth of changed path)` instead of `O(total number of fields in the tree)`. The full diff only runs if computed values actually cascade changes to other parts of the tree.

Tests:

* 11 unit tests for `expandChangedPaths`: ancestor expansion, changesFor filtering, deduplication, nested arrays, edge cases (empty paths, unregistered paths, single-segment paths).
* 12 integration tests for `updateModel` with `changedPaths`: verifies identical results (model data, validation errors) with and without hints for computed values, validations, conditions, array items, multiple simultaneous changed paths, cascading computed values across multiple passes, and backward compatibility with the legacy `CompareFn` signature.

### CODE REVIEW

File: packages/yavl/src/validate/updateModel.ts [hunk 1]
```diff
@@ -4,33 +4,44 @@ import notifySubscribers from './notifySubscribers';
...
-    isEqualFn?: CompareFn,
+    isEqualFnOrOptions?: CompareFn | UpdateModelOptions,
   ): void;
 }
 const updateModel: UpdateModelFn = <Data, ExternalData = undefined, ErrorType = string>(
   context: ModelValidationContext<Data, ExternalData, ErrorType>,
   data: Data,
   externalData?: ExternalData,
-  isEqualFn: CompareFn = Object.is,
+  isEqualFnOrOptions?: CompareFn | UpdateModelOptions,
```
Comment (xmahle) [inline packages/yavl/src/validate/updateModel.ts]: Maybe separate these parameters... so that they could be used at the same time..

Comment (ledancs) [reply packages/yavl/src/validate/updateModel.ts]: addressed ✅

**Review by xmahle — APPROVED** [review]
